### PR TITLE
New Exercise: Acronym

### PR DESCRIPTION
**Pull request by Stargator**
Implement [specification from problem-specifications](https://github.com/exercism/problem-specifications/tree/master/exercises/acronym)

@exercism/dart

### CODE REVIEW

**Comment by Stargator** [comment]
The specification for this exercise was updated to fit a new definition, double check that there aren;t any additional changes

File: exercises/acronym/lib/example.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Acronym {
+  String abbreviate(final String phrase) {
+    StringBuffer acronymBuilder = new StringBuffer();
+
+    RegExp expression = new RegExp(r'(^[A-z])|(\s[A-z])|-[A-z]');
+
+    List<Match> matchList = expression.allMatches(phrase).toList();
+
+    for (Match match in matchList) {
+      acronymBuilder.write(match.group(0).trim().toUpperCase().replaceAll("-", ""));
```
Comment (jvarness) [inline exercises/acronym/lib/example.dart]: If we were able to use the split to get rid of the characters that aren't alpha characters, the `replaceAll` may no longer be necessary.

File: exercises/acronym/lib/example.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Acronym {
+  String abbreviate(final String phrase) {
+    StringBuffer acronymBuilder = new StringBuffer();
+
+    RegExp expression = new RegExp(r'(^[A-z])|(\s[A-z])|-[A-z]');
```
[the remark this answers is not in the document]
Comment (Stargator) [reply exercises/acronym/lib/example.dart]: Sure I'll look at that.

File: config.json [hunk 3]
```diff
@@ -108,11 +108,25 @@
...
+        "equality",
+        "regular_expressions",
+        "strings"
+      ]
+    },
     {
       "slug": "phone-number",
       "uuid": "8649b2b4-02f6-4947-92b9-d1cc0d7c3594",
       "core": false,
-      "unlocked_by": "word-count",
```
Comment (jvarness) [inline config.json]: Does this mean a new exercise needs to be unlocked by word-count other than phone-number?

Comment (Stargator) [reply config.json]: Core exercises **cannot** be unlocked by another, so the order of core exercises matters as far as how the user will progressed through them,

Non-core exercises, I believe, **have to** be unlocked by another exercise. The `configlet` tool will check for anything that doesn't meet these conditions.

File: exercises/acronym/lib/example.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (Stargator) [reply exercises/acronym/lib/example.dart]: I'm having trouble with `.split`. I have the regular expression correct (tested it), but split isn't working as I expected.

[the remark this answers is not in the document]
Comment (Stargator) [reply exercises/acronym/lib/example.dart]: Got it, it wasn't the expression, it was how I was creating an instance of RegExp...I wasn't

**Comment by Stargator** [comment]
Hey @jvarness It's ready for a another review

File: exercises/acronym/lib/example.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class Acronym {
+  String abbreviate(final String input) {
+    StringBuffer acronymBuilder = new StringBuffer();
+
+    RegExp separatorsExp = new RegExp(r'( )|(-)');
```
Comment (jvarness) [inline exercises/acronym/lib/example.dart]: So now we're only splitting on spaces and hyphens?

Should we split on any kind of white space? I think `\w` matches all whitespace.

We don't need to consider any other non-alphanumeric characters as part of this?

Comment (jvarness) [reply exercises/acronym/lib/example.dart]: I guess that makes a lot of sense based on the test cases. We don't have to split on all non-alphanumeric characters.

Although, I'm curious... If any splitted substring started with a non-alphanumeric character... We could see some really weird acronyms.

Comment (Stargator) [reply exercises/acronym/lib/example.dart]: @jvarness If you can find an acronym that starts with that (phrase, company name, etc), then we can certainly add that as a test case. :)

**Pull request merged**